### ISSUE TITLE
[panic] Concurrent map iteration and write

### ISSUE THREAD

**Issue by dsprenkels** [mrngm/rufs#5]
```
sep 10 22:04:12 roku rufs[2840]: fatal error: concurrent map iteration and map write
sep 10 22:04:12 roku rufs[2840]: goroutine 7 [running]:
sep 10 22:04:12 roku rufs[2840]: runtime.throw(0x86b5ca, 0x26)
sep 10 22:04:12 roku rufs[2840]:         /usr/lib/go-1.10/src/runtime/panic.go:616 +0x81 fp=0xc42005fd50 sp=0xc42005fd30 pc=0x42b341
sep 10 22:04:12 roku rufs[2840]: runtime.mapiternext(0xc42005fee0)
sep 10 22:04:12 roku rufs[2840]:         /usr/lib/go-1.10/src/runtime/hashmap.go:747 +0x55c fp=0xc42005fde0 sp=0xc42005fd50 pc=0x40a59c
sep 10 22:04:12 roku rufs[2840]: main.(*Server).fsScanner(0xc420160870, 0xc420194000)
sep 10 22:04:12 roku rufs[2840]:         /home/daan/go/src/github.com/Jille/rufs/server.go:244 +0x2d1 fp=0xc42005ffd0 sp=0xc42005fde0 pc=0x765dd1
sep 10 22:04:12 roku rufs[2840]: runtime.goexit()
sep 10 22:04:12 roku rufs[2840]:         /usr/lib/go-1.10/src/runtime/asm_amd64.s:2361 +0x1 fp=0xc42005ffd8 sp=0xc42005ffd0 pc=0x457c01
sep 10 22:04:12 roku rufs[2840]: created by main.(*Server).Run
sep 10 22:04:12 roku rufs[2840]:         /home/daan/go/src/github.com/Jille/rufs/server.go:122 +0xb0
[snip]
```

**Comment by dsprenkels** [comment]
Only triggers when `hashcache.dat` exists during startup.

**Comment by mrngm** [comment]
Reads and writes on a `map` must be surrounded by some form of locking, such as `sync.Mutex`.

**Comment by dsprenkels** [comment]
Fixed in #5.
